app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -66,38 +198,46 @@ export default class HighlighterRects extends Component {
 
   render () {
     const {source, screenshotInteractionMode, containerEl, searchedForElementBounds,
-           isLocatorTestModalVisible, scaleRatio} = this.props;
+           isLocatorTestModalVisible, scaleRatio, centroidMode} = this.props;
 
-    // Recurse through the 'source' JSON and render a highlighter rect for each element
     const highlighterRects = [];
-
+    const highlighterCentroids = [];
+    const elements = this.getElements(source);
     let highlighterXOffset = 0;
+    let screenshotEl = null;
+
     if (containerEl) {
-      const screenshotEl = containerEl.querySelector('img');
+      screenshotEl = containerEl.querySelector('img');
       highlighterXOffset = screenshotEl.getBoundingClientRect().left -
                            containerEl.getBoundingClientRect().left;
     }
 
-    let recursive = (element, zIndex = 0) => {
-      if (!element) {
-        return;
+    let renderElements = (source) => {
+      for (let elem of source) {
+        highlighterRects.push(<HighlighterRect {...this.props}
+          dimensions={elem.properties}
+          element={elem.element}
+          scaleRatio={scaleRatio}
+          key={elem.properties.path}
+          xOffset={highlighterXOffset}
+        />);
       }
-      highlighterRects.push(<HighlighterRect {...this.props}
-        element={element}
-        zIndex={zIndex}
-        scaleRatio={scaleRatio}
-        key={element.path}
-        xOffset={highlighterXOffset}
-      />);
-
-      if (element.children) {
-        for (let childEl of element.children) {
-          recursive(childEl, zIndex + 1);
-        }
+    };
+
+    let renderCentroids = (centroids) => {
+      for (let elem of centroids) {
+        highlighterCentroids.push(<HighlighterCentroid {...this.props}
+          currentRender={elem.type}
+          elementProperties={elem.properties}
+          element={elem.element}
+          scaleRatio={scaleRatio}
+          key={`C:${elem.properties.path}`}
```

Comment:
it looks like `key` is not retrieved in this component and is always overwritten, so do we need to include it here?